omnigibson/robots/behavior_robot.py

```diff
@@ -546,27 +574,68 @@ def load(self) -> None:
             self.scene.add_object(self.ghost_hand)
 
     @property
-    def local_position_orientation(self) -> Tuple[Iterable[float], Iterable[float]]:
+    def local_position_orientation(
+        self,
+    ) -> Tuple[Iterable[float], Iterable[float]]:
         """
         Get local position and orientation w.r.t. to the body
         Returns:
             Tuple[Array[x, y, z], Array[x, y, z, w]]
 
         """
-        return T.relative_pose_transform(*self.get_position_orientation(), *self.parent.get_position_orientation())
+        import warnings
+
+        warnings.warn(
+            "local_position_orientation is deprecated. Use get_position_orientation instead.", DeprecationWarning
+        )
+        return self.get_position_orientation(frame="parent")
 
-    def get_position_orientation(self) -> Tuple[Iterable[float], Iterable[float]]:
+    def get_position_orientation(
+        self, frame: Literal["world", "scene", "parent"] = "world"
+    ) -> Tuple[Iterable[float], Iterable[float]]:
         """
-        Get position and orientation in the world space
+        Gets robot's pose with respect to the specified frame.
+
+        Args:
+            frame (Literal): frame to get the pose with respect to. Default to world. parent frame
+            get position relative to the object parent. scene frame get position relative to the scene.
+
         Returns:
-            Tuple[Array[x, y, z], Array[x, y, z, w]]
+            2-tuple:
+                - 3-array: (x,y,z) position in the specified frame
+                - 4-array: (x,y,z,w) quaternion orientation in the specified frame
         """
-        return self._root_link.get_position_orientation()
 
-    def set_position_orientation(self, pos: Iterable[float], orn: Iterable[float]) -> None:
+        if frame == "world" or frame == "scene":
+
+            position, orientation = self._root_link.get_position_orientation()
+
+            if frame == "scene" and self.scene is not None:
+                position, orientation = T.relative_pose_transform(
+                    position, orientation, *self.parent.get_position_orientation()
+                )
+
+            return position, orientation
+
+        else:
+
+            return T.relative_pose_transform(*self.get_position_orientation(), *self.parent.get_position_orientation())
```

Comment:
not sure this is correct - we should sit down and make sure the BehaviorRobot is implemented correctly and use VR to do some end2end test